packages/maskbook/src/extension/background-script/EthereumServices/providers/MetaMask.ts

```diff
@@ -63,6 +71,19 @@ export async function requestAccounts() {
     return accounts
 }
 
+export async function popupMetaMaskUnlocked() {
+    const isUnlocked = await detectIfMetaMaskUnlocked()
+    if (!isUnlocked) {
+        await provider!.request({ method: 'eth_requestAccounts', params: [] })
+    }
+}
+
+async function detectIfMetaMaskUnlocked(accounts: string[] = []) {
+    const isUnlocked = Boolean(await provider!._metamask?.isUnlocked()) || accounts.length > 0
```

Comment:
As the api doc indicates, `provider._metamask.isUnlocked()` is still an experimental methods, I find it really returns wrong result after I press password to unlocked the mm, which trigger the `accountsChanged` event, it still returns false: 

![image](https://user-images.githubusercontent.com/63733714/98634999-2edc8000-235f-11eb-93b9-bbbb80612783.png)

So, I add one more check: when we can get non-empty accounts, mm is unlocked.